src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -172,3 +172,11 @@ def _lock_cb(cls, mode, n, file, line):
                     mode, n, file, line
                 )
             )
+
+
+# OpenSSL is not thread safe until the locks are initialized. We call this
+# method in module scope so that it executes with the import lock. On
+# Pythons < 3.4 this import lock is a global lock, which can prevent a race
+# condition registering the OpenSSL locks. On Python 3.4+ the import lock
+# is per module so this approach will not work.
```

Comment:
If that doesn't work, does any of this work?

On Mon, Oct 26, 2015 at 9:42 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/bindings/openssl/binding.py
> <https://github.com/pyca/cryptography/pull/2446#discussion_r43076455>:
>
> > @@ -172,3 +172,11 @@ def _lock_cb(cls, mode, n, file, line):
> >                      mode, n, file, line
> >                  )
> >              )
> > +
> > +
> > +# OpenSSL is not thread safe until the locks are initialized. We call this
> > +# method in module scope so that it executes with the import lock. On
> > +# Pythons < 3.4 this import lock is a global lock, which can prevent a race
> > +# condition registering the OpenSSL locks. On Python 3.4+ the import lock
> > +# is per module so this approach will not work.
>
> It didn't seem to work for people who were testing, but it won't hurt so
> maybe we should still do it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2446/files#r43076455>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084